src/DotVVM.Framework/Runtime/DefaultDotvvmViewBuilder.cs

```diff
@@ -70,11 +70,8 @@ protected void VerifySpaRequest(IDotvvmRequestContext context, DotvvmView page)
             if (context.IsSpaRequest)
             {
                 var spaContentPlaceHolders = page.GetAllDescendants().OfType<SpaContentPlaceHolder>().ToList();
-                if (spaContentPlaceHolders.Count > 1)
-                {
-                    throw new Exception("Multiple controls of type <dot:SpaContentPlaceHolder /> found on the page! This control can be used only once!");   // TODO: exception handling
-                }
-                if (spaContentPlaceHolders.Count == 0 || spaContentPlaceHolders[0].GetSpaContentPlaceHolderUniqueId() != context.GetSpaContentPlaceHolderUniqueId())
+
+                if (spaContentPlaceHolders.Count == 0 || !context.GetSpaContentPlaceHolderUniqueId().Contains(spaContentPlaceHolders[0].GetSpaContentPlaceHolderUniqueId()))
```

Comment:
We should split the ID by the separator and then call `Contains`. Maybe, we could even use a proper array, we do not care too much about backwards compatibility in the DotVVM protocol.